### PR TITLE
Small perf improvements

### PR DESCRIPTION
**Pull request by Ch0ronomato**
## Description
There was a warning about copying a tensor, this change removes that. In addition, until we decide to keep torch autograd going around, I disabled it here to free up a bit of memory during the runtime. This brought back about 80us back from the mean time in the best case. 

## Related Issue

- [ ] Closes #
- [ ] Related to #1110 

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pytensor--1128.org.readthedocs.build/en/1128/

### CODE REVIEW

**Comment by ricardoV94** [comment]
Torch autograd is on by default :O?

**Comment by ricardoV94** [comment]
Is it though? https://dev-discuss.pytorch.org/t/how-does-torch-compile-work-with-autograd/1621/14

File: pytensor/link/pytorch/dispatch/basic.py [hunk 1]
```diff
@@ -123,7 +123,10 @@ def arange(start, stop, step):
 def pytorch_funcify_Join(op, **kwargs):
     def join(axis, *tensors):
         # tensors could also be tuples, and in this case they don't have a ndim
-        tensors = [torch.tensor(tensor) for tensor in tensors]
+        tensors = [
```
Comment (ricardoV94) [inline pytensor/link/pytorch/dispatch/basic.py:126]: I don't think this comprehension is needed at all when I tested in my PR

Comment (ricardoV94) [reply pytensor/link/pytorch/dispatch/basic.py:126]: https://github.com/pymc-devs/pytensor/pull/1118

**Comment by Ch0ronomato** [comment]
> Is it though? https://dev-discuss.pytorch.org/t/how-does-torch-compile-work-with-autograd/1621/14

Ah - no it's not. When I did the autograd experiment, I had to turn it on for the tensors we cared about. We could even close this up

**Comment by Ch0ronomato** [comment]
I'm gonna close this - I don't think there is much being changed given torch autograd is off in our case by default.

**Pull request closed without merging**